store/sql_user_store.go

```diff
@@ -510,10 +512,12 @@ func (s SqlUserStore) GetEtagForDirectProfiles(userId string) StoreChannel {
 			                AND Category = 'direct_channel_show')
 			ORDER BY UpdateAt DESC LIMIT 1
         `, map[string]interface{}{"UserId": userId})
-		if err != nil {
-			result.Data = fmt.Sprintf("%v.%v.%v.%v", model.CurrentVersion, model.GetMillis(), utils.Cfg.PrivacySettings.ShowFullName, utils.Cfg.PrivacySettings.ShowEmailAddress)
+
+		if err != nil || len(ids) == 0 {
+			result.Data = fmt.Sprintf("%v.%v.0.%v.%v", model.CurrentVersion, model.GetMillis(), utils.Cfg.PrivacySettings.ShowFullName, utils.Cfg.PrivacySettings.ShowEmailAddress)
 		} else {
-			result.Data = fmt.Sprintf("%v.%v.%v.%v", model.CurrentVersion, updateAt, utils.Cfg.PrivacySettings.ShowFullName, utils.Cfg.PrivacySettings.ShowEmailAddress)
+			allIds := strings.Join(ids, "")
+			result.Data = fmt.Sprintf("%v.%v.%v.%v.%v", model.CurrentVersion, md5.Sum([]byte(allIds)), len(ids), utils.Cfg.PrivacySettings.ShowFullName, utils.Cfg.PrivacySettings.ShowEmailAddress)
```

Comment:
Won't `allIds` always just contain a single id since the SQL query has `LIMIT 1`?